solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -368,8 +368,15 @@ public void command() {
               }
 
               cc.cancelCoreRecoveries();
-
-              registerAllCoresAsDown(registerOnReconnect, false);
+              
+              try {
+                registerAllCoresAsDown(registerOnReconnect, false);
+              } catch (SessionExpiredException e) {
+                // zk has to reconnect
+                throw e;
+              } catch (Exception e) {
+                log.warn("Exception while trying to register all cores as DOWN", e);
```

Comment:
don't you want to re-throw the exception here?

Reply:
I'll add a better comment to the code.

Reply:
I should've guessed :) I just thought that you missed handling more cases.